independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java

```diff
@@ -17,6 +18,8 @@
 
     static final Logger LOGGER = Logger.getLogger(Decorators.class);
 
+    private static final Set<DotName> builtInBeanTypes = Set.of(DotNames.BEAN_MANAGER, DotNames.INSTANCE, DotNames.EVENT);
```

Comment:
What about other built-in beans like `Bean<T>`, `@Intercepted Bean<?>`, `@Decorated Bean<?>`, `Interceptor<T>`, `Decorator<T>`,  `InjectionPoint`,  `RequestContextController`, `InterceptionFactory` ? (Maybe more, these are just the ones I remember off the top of my head)

Reply:
I am well aware of them and I was planning to discuss this with @mkouba once he gets back from PTO.
I only added these three since they are "most common" candidates for this scenario. I am not sure the other built in beans would make much sense with decorators.

Reply:
@mkouba do you think we should make an exhaustive list of all built-in beans or keep it as is?
Personally, I don't see anyone trying to decorate anything else than BM, event and instance but I might be wrong.

Reply:
> do you think we should make an exhaustive list of all built-in beans or keep it as is?

I think that we could just reuse the `io.quarkus.arc.processor.BuiltinBean` enum and `BuiltinBean.getRawTypeDotNames()` but exclude the `BuiltinBean.RESOURCE`.